src/objects.c

```diff
@@ -809,10 +809,13 @@ void (*MakeImmutableObjFuncs[LAST_REAL_TNUM+1])( Obj );
 void MakeImmutable(Obj obj)
 {
     if (IS_MUTABLE_OBJ(obj)) {
-        SET_OBJ_FLAG(obj, OBJ_FLAG_IMMUTABLE);
+        if (!IS_INTOBJ(obj) && !IS_FFE(obj)) {
```

Comment:
I think this is a consequence of me introducing a bug in IS_MUTABLE_OBJ (which I also introduced) which meant that SET_OBJ_FLAG would be called on `FFE`s or `INTOBJ`s. This is something I fixed on the train yesterday (but didn't push yet, since I break HPC-GAP in interesting ways which I want to understand first).